src/pybind/mgr/prometheus/module.py

```diff
@@ -432,15 +439,54 @@ def _setup_static_metrics(self):
                 'Number of {} objects'.format(state),
             )
 
+        for check in HEALTH_CHECKS:
+            path = 'healthcheck_{}'.format(check['name'].lower())
+            metrics[path] = Metric(
+                'gauge',
+                path,
+                check['description'],
+            )
+
         return metrics
 
     @profile_method()
     def get_health(self):
+
+        def _get_value(message, delim=" ", word_pos=0):
+            """Extract value from message (default is 1st field)"""
+            v_str = message.split(delim)[word_pos]
+            if v_str.isdigit():
+                return int(v_str), 0
+            return 0, 1
+
         health = json.loads(self.get('health')['json'])
+        # set overall health
         self.metrics['health_status'].set(
             health_status_to_number(health['status'])
         )
 
+        # Examine the health to see if any health checks triggered need to
+        # become a metric.
+        active_healthchecks = health.get('checks', {})
+        active_names = active_healthchecks.keys()
+
+        # healthcheck metrics must always be present in the scrape
+        for healthcheck in HEALTH_CHECKS:
+            v = 0
+            err = 0
+
+            check_name = healthcheck['name']
+            if check_name in active_names:
+                check_data = active_healthchecks[check_name]
+                message = check_data['summary'].get('message', '')
+                if check_name == "SLOW_OPS":
+                    v, err = _get_value(message)
+                if err:
+                    self.log.warning("healthcheck {} message format has changed and is incompatible".format(check_name))
```

Comment:
`continue` in this case?

Reply:
Yes - I don't thing an alert format change warrants an exception.

Reply:
parse failure renders the `v` invalid. it does not make sense to collect it using `self.metrics[path].set(v)`.

Reply:
I think what you're saying is if the parse fails, we could issue a warning to the log - but don't emit a metric if the parsing fails...right? that way the metric is missing not 0?

Reply:
right. i think missing is better than "0". the former is a signal of error and implies "i don't know!", the latter means, "it's 0". where "0" is far from a warning in this context.